src/components/dropdownSearch/DropdownSearchReducers.tsx

```diff
@@ -111,11 +154,17 @@ export const dropdownSearchReducer = (state: IDropdownSearchState = dropdownSear
         return {
           ...state,
           id: action.payload.id,
-          selectedOption: state.activeOption,
           isOpened: false,
+          selectedOptions: state.selectedOptions.push(state.activeOption),
```

Comment:
it returns a new FixedQueue object yes, but it keeps your first `instantiated this.queue` as a reference underneath all the time, in other words, you're not really creating a new queue, just a new reference to it.